### PR TITLE
DM-35063: continue deprecation and removal of integer dataset IDs

### PR DESCRIPTION
**Pull request by TallJimbo**
## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: tests/test_butler.py [hunk 1]
```diff
@@ -2083,23 +2083,6 @@ def testTransferUuidToUuid(self):
...
-    def testTransferIntToInt(self):
-        with self.assertWarns(FutureWarning):
-            self.create_butlers(
-                "lsst.daf.butler.registry.datasets.byDimensions.ByDimensionsDatasetRecordStorageManager",
-                "lsst.daf.butler.registry.datasets.byDimensions.ByDimensionsDatasetRecordStorageManager",
-            )
-        # int dataset ID only allows UNIQUE
-        self.assertButlerTransfers()
-
-    def testTransferIntToUuid(self):
```
Comment (timj) [inline tests/test_butler.py:2095]: If we are allowing migration for the moment, shouldn't we still be testing int to UUID? Presumably the problem is that we can't any longer create a test repo from scratch? The only alternative is to have a pre-created repo for the test but it's likely not worth it.

File: tests/test_simpleButler.py [hunk 2]
```diff
@@ -49,9 +49,11 @@ class SimpleButlerTestCase(unittest.TestCase):
     can instead utilize an in-memory SQLite Registry and a mocked Datastore.
     """
 
-    datasetsManager = "lsst.daf.butler.registry.datasets.byDimensions.ByDimensionsDatasetRecordStorageManager"
-    datasetsImportFile = "datasets.yaml"
```
Comment (timj) [inline tests/test_simpleButler.py:53]: Does this mean that this file is no longer used?

**Review by timj — APPROVED** [review]
Looks okay.

File: tests/test_butler.py (hunk 1 above)
Comment (TallJimbo) [reply tests/test_butler.py:2095]: You've replicated my train of thought on this exactly.

File: tests/test_simpleButler.py (hunk 2 above)
Comment (TallJimbo) [reply tests/test_simpleButler.py:53]: I've left in place the test that imports this file (with integer IDs) into a UUID data repository.  I don't know if we want to support that indefinitely, but I haven't tried to deprecate it (I didn't think about it at all when writing the RFC).

**Pull request merged**